docs/modules/language-reference/pages/index.adoc

```diff
@@ -3179,6 +3179,33 @@ This section discusses language features that are generally more relevant to tem
 <<blank-identifiers,Blank Identifiers>> +
 <<projects,Projects>>
 
+[[the-meaning-of-new]]
+=== The Meaning of `new`
+
+Pkl has no "true" object instantiation: the `new` keyword is a special case of <<amending-objects, amending>>.
+In Pkl, objects can be instantiated in three major ways:
+
+* Amending a property/entry `someProperty {}` directly, which may specify properties that have not previously been declared or defined
+* Invoking `new` without type information `new {}`, which amends a contextually inferred value
+* Invoking `new` with explicit type information: `new Dynamic {}`, which always amends the default value of the specified type
+
+When amending previously undeclared properties and using the untyped `new` invocation, there is no explicit parent value to amend.
+In these cases, Pkl infers the amend operation's parent value based on context.
+In order of precedence:
+
+1. When amending an existing value, the parent value's type is used.
```

Comment:
Nit picking, but the parent value's type is never used. It's just the value itself that gets amended.